R/lassnags.r

```diff
@@ -0,0 +1,246 @@
+#
+# lidR is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+# ===============================================================================
+
+#' Snag classification of airborne LiDAR point clouds using neighborhood metrics
+#'
+#' Snag clasification using the algorithim proposed by Wing et al 2015 (see details). The function
+#' attributes to each point of the point cloud a number identifying a snag class (\code{snagCls} column).
+#' By default the classification is done at the point cloud level and uses LiDAR intensity thresholds and 
+#' specified neighborhoods to differentiate bole and branch from foilage points. There is currently only 
+#' one algorithms implemented.
+#'
+#' @param las An object of the class \code{LAS}
+#' @param algorithm character. The name of an algorithm. At present, can only be \code{"wing2015"}.
+#' (see sections relevant to each
+#' algorithm).
+#' @param ... parameters for the algorithms. These depend on the algorithm used (see details
+#' about the algorithm)
+#' 
+#' @return Nothing, the point cloud is updated by reference. A new field is added to the las file (\code{SnagCls}),
+#' for each point with the corresponding classificaiton: 0) live tree, 1) general snag, 2) small snag, 3) live 
+#' crown edge snag, and 4) high canopy cover snag.
+#'
+#' @section Wing et al. 2016:
+#' This is an automated filtering algorithm that utilizes three dimensional neighborhood lidar point-based
+#' intensity and density statistics to remove lidar points associated with live trees and retain lidar points
+#' associated with snags devleoped by Wing et al (2015 - see references). 
+#' Note that this algorithm strictly performs a classifciation based on user input while the original publicaiton's 
+#' methods also included a segmentation step and some pre- (filtering for first and single returns only) and post-
+#' process (filtering for only the snag classified points prior to segmentation) tasks whihc are now expected to 
+#' be preformed by the user. Also, this implemntation may have some differences compared with the original method
+#' due to potential mis-interpretation of the Wing et al. manuscript, specifically Table 2 where they present four
+#' groups of conditional assessments with their required neighborhood point density and average BBPR values (BBPR 
+#' = branch and bole point ratio; PDR = point density requirement). 
+#' 
+#' \describe{
```

Comment:
This the old way I document `lastress`. This was not good. Look at how I do that now in the devel branch.